src/MatrixClientPeg.ts

```diff
@@ -275,14 +275,17 @@ class MatrixClientPegClass implements IMatrixClientPeg {
 
         if (SettingsStore.getValue("feature_sliding_sync")) {
             const proxyUrl = SettingsStore.getValue("feature_sliding_sync_proxy_url");
+            const wellKnownProxyUrl = this.safeGet().getClientWellKnown()?.["org.matrix.msc3575.proxy"]?.url;
             if (proxyUrl) {
                 logger.log("Activating sliding sync using proxy at ", proxyUrl);
+            } else if (wellKnownProxyUrl) {
+                logger.log("Activating sliding sync using proxy at ", wellKnownProxyUrl);
             } else {
                 logger.log("Activating sliding sync");
             }
             opts.slidingSync = SlidingSyncManager.instance.configure(
                 this.matrixClient,
-                proxyUrl || this.matrixClient.baseUrl,
+                proxyUrl || wellKnownProxyUrl || this.matrixClient.baseUrl,
```

Comment:
I can't get this to work.

Reply:
This code looks racy, `getClientWellKnown` will return `undefined` if the request has not yet completed. Likely you want `await this.safeGet().waitForClientWellKnown()?...`

Reply:
Thank you! I'm much closer!

However, if I change to use `waitForClientWellKnown()`, it throws a `Unable to load session Error: Client is not running` error. I guess the client isn't fully initialised at this point in the function?

Reply:
You can probably use `fetchClientWellKnown` instead then

Reply:
Yesssss! It's working! However, it does require changing the `fetchClientWellKnown` function's scope as it's currently `private`.

Alternatively, could [the if statement which calls `fetchClientWellKnown`](https://github.com/matrix-org/matrix-js-sdk/blob/0ff0093380b7f955d50ebed0648358a9b5b06c4e/src/client.ts#L1531-L1536) be moved further up the function?

I guess either way requires its own PR on the `matrix-js-sdk`?

Reply:
Ah good point, probably you can call something like `AutoDiscovery.getRawClientConfig(this.getDomain() ?? undefined)` directly, or make a PR to make the method public, up to you.